test/functionalTest/cases/3190_notify_only_attributes_that_change/notify_only_attributes_that_change_with_cond_empty_notify.test

```diff
@@ -0,0 +1,438 @@
+# Copyright 2019 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Notify only attributes that change for non-empty condition.attrs and empty notification.attrs for NGSIv2 POST /v2/entities, POST /v2/subscriptions, Dump and reset accumulator, PATCH /v2/entities/Room1/attrs, Dump and reset accumulator, PATCH /v2/entities/Room1/attrs, Dump and reset accumulator, PATCH /v2/entities/Room1/attrs, Dump and reset accumulator, PATCH /v2/entities/Room1/attrs, Dump and reset accumulator.
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print localhost $LISTENER_PORT
+
+--SHELL--
+
+#
+# 01. POST /v2/entities, create Room1 entity with A, B, C, and D attributes.
+# 02. POST /v2/subscriptions, to create subscription for Room1 with cond attrs: [A, B, C] notify attrs: [], onlyChangedAttrs: true.
+# 03. Dump and reset accumulator, see A, B, C and D (as this is initial notification).
+# 04. PATCH /v2/entities/Room1/attrs, update the value of D attribute.
+# 05. Dump and reset accumulator, see no notification.
+# 06. PATCH /v2/entities/Room1/attrs, update the value of A and B attribute.
+# 07. Dump and reset accumulator, see A and B.
+# 08. PATCH /v2/entities/Room1/attrs, update the value of A and D attribute.
+# 09. Dump and reset accumulator, see A and D.
+# 10. PATCH /v2/entities/Room1/attrs, update the value of A, B, C and D attribute.
+# 11. Dump and reset accumulator, see A, B and C.
```

Comment:
D should be also updated in this case due it matches both:

* It has changed
* Notify attrs is empty, so no attribute is being filtered out in notifications

Reply:
In fact, it seems that is being included (around line 421 in this file) so probably is just a typo in this line comment.

Reply:
fixed in 1a364d6